index.test.js

```diff
@@ -146,6 +149,7 @@ const urls = [
 	'https://www.npmjs.com/packaasdge/node',
 	'https://example.com/nodejs/node/blob/cc8fc46/.gitignore',
 	'https://example.site/한글로-된-URL',
+	'https://한글로-된-경로.com/하위경로#한글-해시',
```

Comment:
I think this is failing because of this comparison, so look here:

https://github.com/refined-github/shorten-repo-url/blob/7c05f2c24eda58c4b87ea88e58b3ae1ed1be3073/index.js#L303-L312

Reply:
I think that instead of using safeDecode, it should use `new URL(getLinkHref(a)).href === new URL(a.textContent).href`. This function can normalize the URL without passing it through `safeDecode`

The catch is that `new URL()` will throw when `a.textContent` is not a URL, so you still need a `safe` function.

Try just changing to this:

```js
function safeDecode(url) {
	try {
		return new URL(url).href;
	} catch {
		return url;
	}
}
```

Reply:
done in 365b3eafa4059407e18c14dac2ee559ef3dcb38f